### PR TITLE
Fix #5779: Use correct id to fetch price and price history

### LINKED ISSUE #5779: Wallet Bug: User visible asset doesn't display correct fiat value in Solana
image

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Use token.symbol for fetching balance
Use token.coingeckoId to fetch price and price history if token.coingeckoId is not empty, otherwise use token.symbol

This pull request fixes #5779 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test Plan:
Please refer to the issue

## Screenshots:

simulator_screenshot_3F58CB2C-946E-4D5D-92B1-064D7575546D | simulator_screenshot_DAC69E83-7FC5-45B4-81D7-AB8219041371
-- | --

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveWallet/Extensions/BraveWalletExtensions.swift [hunk 1]
```diff
@@ -194,3 +194,9 @@ extension BraveWallet.NetworkInfo {
     && coin == token.coin
   }
 }
+
+extension BraveWallet.BlockchainToken {
+  var assetRatioId: String {
```
Comment (StephenHeaps) [inline BraveWallet/Extensions/BraveWalletExtensions.swift:200]: nit: can we add a small comment for what this is used for

**Review by StephenHeaps — COMMENTED** [review]
Nice 🙂, looks like we may need to update calls in `TransactionConfirmationStore` and `TransactionDetailsStore` to use new `assetRatioId` too?

File: BraveWallet/Crypto/Stores/PortfolioStore.swift [hunk 2]
```diff
@@ -196,10 +198,10 @@ public class PortfolioStore: ObservableObject {
   
   /// Fetches the prices for a given list of symbols, giving a dictionary with the price for each symbol
```
Comment (StephenHeaps) [inline BraveWallet/Crypto/Stores/PortfolioStore.swift]: nit: maybe should update `symbol` in comment to `priceId` / `assetRatioId`.

File: BraveWallet/Crypto/Stores/PortfolioStore.swift [hunk 3]
```diff
@@ -209,19 +211,19 @@ public class PortfolioStore: ObservableObject {
   
   /// Fetches the price history for the given symbols, giving a dictionary with the price history for each symbol
```
Comment (StephenHeaps) [inline BraveWallet/Crypto/Stores/PortfolioStore.swift]: Similar nit about function comment

**Review by StephenHeaps — APPROVED** [review]
👍

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**